packages/nice-grpc-client-middleware-devtools/src/index.ts

```diff
@@ -0,0 +1,180 @@
+import { isAbortError } from 'abort-controller-x';
+import { Message } from 'google-protobuf';
+import {
+  CallOptions,
+  ClientError,
+  ClientMiddleware,
+  ClientMiddlewareCall,
+  composeClientMiddleware,
+} from 'nice-grpc-web';
+
+export const devtoolsUnaryLoggingMiddleware: ClientMiddleware = async function* devtoolsLoggingMiddleware<Request, Response>(
+  call: ClientMiddlewareCall<Request, Response>,
+  options: CallOptions
+): AsyncGenerator<Response, Response | void, undefined> {
+  // skip streaming calls
+  if (call.requestStream || call.responseStream) {
+    return yield* call.next(call.request, options);
+  }
+
+  const { path } = call.method;
+
+  let reqObj: any = undefined;
+
+  // TODO: When would it not be a jspb.Message?
+  if (call.request instanceof Message) {
+    const reqMsg = call.request as Message;
+    reqObj = reqMsg.toObject();
+  } else {
+    //eslint-disable-next-line
+    reqObj = call.request;
+    // console.warn('TODO: Request isn\'t a jspb.Message');
+  }
```

Comment:
I suggest to extract this code to a serialization function.

That function may check whether the value has a `toObject` method, without relying on `instanceof` which is fragile and pulls in the `google-protobuf` dependency even if it is not used.